apps/web/src/hooks/useInstrumentVisualization.ts

```diff
@@ -195,19 +218,42 @@ export function useInstrumentVisualization({ params }: UseInstrumentVisualizatio
   };
 
   useEffect(() => {
-    if (recordsQuery.data) {
-      const records: InstrumentVisualizationRecord[] = [];
-      for (const record of recordsQuery.data) {
-        const props = record.data && typeof record.data === 'object' ? record.data : {};
-        records.push({
-          __date__: record.date,
-          __time__: record.date.getTime(),
-          ...record.computedMeasures,
-          ...props
+    const fetchRecords = () => {
+      try {
+        const sessions = sessionsUsernameQuery.data;
+        if (recordsQuery.data && sessions) {
+          // Fetch all sessions in parallel
+
+          // Build records with looked-up data
+          const records: InstrumentVisualizationRecord[] = recordsQuery.data.map((record) => {
+            const props = record.data && typeof record.data === 'object' ? record.data : {};
+            const usersSession = sessions.find((s) => s.id === record.sessionId);
+
+            const username = usersSession?.user?.username ?? 'N/A';
+
+            return {
+              __date__: record.date,
+              __time__: record.date.getTime(),
+              username: username,
+              ...record.computedMeasures,
+              ...props
+            };
+          });
+
+          setRecords(records);
+        }
+      } catch (error) {
+        console.error('Error occurred: ', error);
+        notifications.addNotification({
+          message: t({
+            en: 'Error occurred finding records',
+            fr: "Une erreur s'est produite lors de la recherche des enregistrements."
+          }),
+          type: 'error'
         });
       }
-      setRecords(records);
-    }
+    };
+    void fetchRecords();
```

Comment:
or just eliminate entirely. there is no need to have a function that is invoked only one time, immediately after the definition, with no arguments.